InnerEye/ML/SSL/datamodules_and_datasets/transforms_utils.py

```diff
@@ -0,0 +1,280 @@
+        self.max_horizontal_shift = config.augmentation.random_affine.max_horizontal_shift
+        self.max_vertical_shift = config.augmentation.random_affine.max_vertical_shift
+        self.max_shear = config.augmentation.random_affine.max_shear
+
+    def transform(self, x: Any) -> Any:
+        return torchvision.transforms.RandomAffine(degrees=self.max_angle,
+                                                   translate=(self.max_horizontal_shift, self.max_vertical_shift),
+                                                   shear=self.max_shear)(x)
+
+
+class Resize(BaseTransform):
+    def __init__(self, config: CfgNode) -> None:
+        super().__init__()
+        self.resize_size = config.preprocess.resize
+
+    def transform(self, x: Any) -> Any:
+        return torchvision.transforms.Resize(self.resize_size)(x)
+
+
+class RandomColorJitter(BaseTransform):
+    def __init__(self, config: CfgNode) -> None:
+        super().__init__()
+        self.max_brightness = config.augmentation.random_color.brightness
+        self.max_contrast = config.augmentation.random_color.contrast
+        self.max_saturation = config.augmentation.random_color.saturation
+
+    def transform(self, x: Any) -> Any:
+        return torchvision.transforms.ColorJitter(brightness=self.max_brightness,
+                                                  contrast=self.max_contrast,
+                                                  saturation=self.max_saturation)(x)
+
+
+class RandomErasing(BaseTransform):
+    def __init__(self, config: CfgNode) -> None:
+        super().__init__()
+        self.scale = config.augmentation.random_erasing.scale
+        self.ratio = config.augmentation.random_erasing.ratio
+
+    def transform(self, x: Any) -> Any:
+        return torchvision.transforms.RandomErasing(p=0.5,
+                                                    scale=self.scale,
+                                                    ratio=self.ratio)(x)
+
+
+class RandomGamma(BaseTransform):
+    def transform(self, image: PIL.Image.Image) -> PIL.Image.Image:
+        gamma = random.uniform(*self.scale)
+        return torchvision.transforms.functional.adjust_gamma(image, gamma=gamma)
+
+    def __init__(self, config: CfgNode) -> None:
+        super().__init__()
+        self.scale = config.augmentation.gamma.scale
+
+
+class ExpandChannels(BaseTransform):
+    """
+    Transforms an image with 1 channel to an image with 3 channels by copying pixel intensities of the image along
+    the 0th dimension.
+    """
+
+    def transform(self, data: torch.Tensor) -> torch.Tensor:
+        return torch.repeat_interleave(data, 3, dim=0)
+
+
+class AddGaussianNoise(BaseTransform):
+
+    def transform(self, data: torch.Tensor) -> torch.Tensor:
+        if np.random.random(1) > self.p_apply:
+            return data
+        noise = torch.randn(size=data.shape) * self.std
+        data = torch.clamp(data + noise, 0, 1)
```

Comment:
maybe 0 and 1 can be defined based on torch.max(data) instead of hard coding it?

Reply:
Mmm here we expect the input image to be in range [0,1], anyways. But i can add an assertion to make sure our input data is really in 0-1

Reply:
update